src/MDMTransitionTarget.m

```diff
@@ -0,0 +1,74 @@
+/*
+ Copyright 2017-present The Material Motion Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDMTransitionTarget.h"
+
+@import MotionTransitioning;
+
+typedef NS_ENUM(NSUInteger, Target) {
+  TargetBack,
+  TargetFore,
+  TargetView,
+};
+
+@implementation MDMTransitionTarget {
+  Target _target;
+  UIView *_targetView; // Valid only if _target == TargetView
+}
+
++ (instancetype)targetWithBackView {
+  MDMTransitionTarget *target = [[[self class] alloc] init];
```

Comment:
Suggestion: all the "target" are a bit confusing.  Maybe this pointer can be named `transitionTarget`?

Reply:
Done.